api/generic-handlers.go

```diff
@@ -62,7 +62,9 @@ func (h corsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Add("Vary", "Origin")
 	origin := r.Header.Get("Origin")
 	if r.Method != "OPTIONS" {
-		if origin == "" {
+		// for post form in browser, the origin maybe 'null'
+		// please refer to https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
```

Comment:
Where is the logic in the reference?
I think the processing of form should be separated from the request.

Reply:
@cannium i have removed that change since it is produced by chrome brower. pls check it again, thanks.